From 66713f77e851c12452729d37e22ef66673852b8f Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 10 Jul 2013 23:50:28 +0200
Subject: [PATCH 2/2] core: uninstall cgroup agent only if we are running
 outside of a container

Since the cgroupfs is currently not virtualized for containers we
shouldn't reset the hosts agent from the container.
---
 src/core/main.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/core/main.c b/src/core/main.c
index efc5791..0ba1d15 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -1943,9 +1943,11 @@ finish:
                         watchdog_close(true);
                 }
 
-                /* avoid the creation of new processes forked by the kernel; at this
-                 * point, we will not listen to the signals anyway */
-                cg_uninstall_release_agent(SYSTEMD_CGROUP_CONTROLLER);
+                /* Avoid the creation of new processes forked by the
+                 * kernel; at this point, we will not listen to the
+                 * signals anyway */
+                if (detect_container(NULL) <= 0)
+                        cg_uninstall_release_agent(SYSTEMD_CGROUP_CONTROLLER);
 
                 execve(SYSTEMD_SHUTDOWN_BINARY_PATH, (char **) command_line, env_block);
                 free(env_block);
-- 
1.8.3.2

